test_conformance/api/test_sub_group_dispatch.cpp

```diff
@@ -75,6 +75,14 @@ int test_sub_group_dispatch(cl_device_id deviceID, cl_context context, cl_comman
     size_t ret_ndrange2d_flattened;
     size_t ret_ndrange3d_flattened;
 
+    // This test should be skipped in the case that the cl_khr_subgroups
+    // extension is not supported.
+    if (!is_extension_available(deviceID, "cl_khr_subgroups"))
+    {
+        log_info("Skipping since cl_khr_subgroups is not supported on device\n");
```

Comment:
The comment and log message could just refer to subgroups being missing rather than `cl_khr_subgroups` given this is a core feature in 2.1+.

But I think actually using the extension for the check is fine, as that is backwards compatible to older versions.

Reply:
I agree. You cannot skip this test in case extension not found on devices 2.1+
In my opinion it should be similar to [this](https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_conformance/subgroups/main.cpp#L66). Shortly:
devices 2.1 and 2.2 - always test (even no extension)
device 3.X - check if subgroups feature is on CL_DEVICE_MAX_NUM_SUB_GROUPS > 0

Reply:
Actually @gwawiork is correct, the specification does not require reporting `cl_khr_subgroups` even when subgroups are supported.  Thinking about it, an implementation **must not** report it if subgroups are implemented without IFP.

Reply:
If you are interested more in subgroups topic I am preparing changes #682  that maybe add some more clarification in subgroups
@alycm When this will be ready and merged then we could I think thick the box in #773

Reply:
We really need more framework support to make these checks easy and reusable. I've created https://github.com/KhronosGroup/OpenCL-CTS/issues/792 to cover this.

Reply:
Agree this won't work for the core feature BTW.